openmdao/recorders/hdf5_recorder.py

```diff
@@ -63,6 +65,8 @@ def record_metadata(self, group):
 
         group = f.require_group('metadata')
 
+        group.create_dataset('om_case_version', data = om_case_version)
```

Comment:
I'm not a fan of the name 'om_case_version' for this.  There are various things in the file that we might change, and the format of a case is only one of them, i.e, at some point we might add the openmdao dependency graph to the file or whatever.  Is there any reason why we can't use something more general, like 'version' or 'format_version' or something like that?  And is 'om' supposed to be shorthand for openmdao?  This version has nothing to do with the version number of openmdao, so adding 'om' to it just seems like it might cause confusion.